internal/controller/grafanaorganization_controller.go

```diff
@@ -132,5 +198,79 @@ func (r GrafanaOrganizationReconciler) reconcileDelete(ctx context.Context, graf
 func (r *GrafanaOrganizationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.GrafanaOrganization{}).
+		// Watch for grafana pod's status changes
+		Watches(
+			&v1.Pod{},
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+				k8sClient := mgr.GetClient()
+				var organizations v1alpha1.GrafanaOrganizationList
+
+				err := k8sClient.List(ctx, &organizations)
+				if err != nil {
+					log.FromContext(ctx).Error(err, "failed to list grafana organization CRs")
+					return []reconcile.Request{}
+				}
+
+				// Reconcile all grafana organizations when the grafana pod is recreated
+				requests := make([]reconcile.Request, 0, len(organizations.Items))
+				for _, organization := range organizations.Items {
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name: organization.Name,
+						},
+					})
+				}
+				return requests
+			}),
+			builder.WithPredicates(predicates.GrafanaPodRecreatedPredicate{}),
+		).
 		Complete(r)
 }
+
+func (r *GrafanaOrganizationReconciler) configureGrafana(ctx context.Context) error {
```

Comment:
I would rather have this function named `reconfigureGrafanaRBAC`

```suggestion
// configureGrafana ensures the RBAC configuration is set in Grafana.
func (r *GrafanaOrganizationReconciler) configureGrafana(ctx context.Context) error {
```